DEV: move Pixi workspace to repo root

lucascolley opened the pull request:
Towards gh-21489.

This PR is an exciting one! It unblocks 2 key strands of work:
- automated dependency upgrades for the Pixi workspace, which is the key item remaining in gh-21489
- new developer docs with drastically reduced complexity, using the Pixi workspace

I think now is a good time to do this, having shown Pixi's capabilities across all of our primary CI operating systems in gh-23792, gh-23654, and gh-23700.

On merge of this PR, the following tasks can be noted for follow-up in issues:
- There is still some scope within gh-21489 to convert more CI jobs to use Pixi, e.g. the CircleCI jobs. We are deliberately not aiming for 100% coverage, as some variety may help smoke out additional issues, but some more coverage might still be nice.
- There are more development tasks/environments to add to the Pixi workspace, most obviously for documentation right now, but there are others too that we might want to inherit from https://github.com/rgommers/pixi-dev-scipystack/blob/main/scipy/pixi.toml
	- gh-23848

lucascolley committed:
DEV: move Pixi workspace to repo root

rgommers commented on pixi.toml:
```diff
@@ -158,13 +158,13 @@ blas-devel = "*"
 [feature.build-task.tasks.build]
 cmd = "spin build --setup-args=-Dblas=blas --setup-args=-Dlapack=lapack --setup-args=-Duse-g77-abi=true"
-cwd = "../.."
+cwd = "."
```
I think this is the default and can hence be left out? Saves a bunch of boilerplate lines.

rgommers reviewed (commented):
Thanks Lucas! I think this will work now, and we're probably ready for this. It'd be good to do the contributor docs update in this same PR I think, that will give the non-expert reviewer a much better impression of the benefits.

I thought for a while about how this will affect my own dev setup - it'll make it slightly more cumbersome, because I can now only run tasks for envs that aren't maintained in the `pixi.toml` inside the scipy repo from the dir one level up, and I'll have to work out some scheme for ensuring the two manifests and build dirs won't clash. But that's all doable.

lucascolley commented:
> I can now only run tasks for envs that aren't maintained in the pixi.toml inside the scipy repo from the dir one level up

I think the long-term goal in my eyes is to just move everything that would be used somewhat frequently in-tree.

> I'll have to work out some scheme for ensuring the two manifests and build dirs won't clash. But that's all doable.

Yeah, I think as long as the out-of-tree setup endeavours to avoid name clashes with the in-tree setup, things should still work.

lucascolley committed:
remove `cwd = "."`

[skip ci]

lucascolley committed:
add docs task, add note on Pixi to building from source docs

lucascolley commented:
okay, I added the docs task and added the following note to the docs:

<img width="759" height="518" alt="image" src="https://github.com/user-attachments/assets/b9181f55-9a15-4ae5-a8ba-6aa39f873997" />

ready to go @rgommers 👀?

tylerjereddy commented on doc/source/building/index.rst:
```diff
@@ -290,6 +290,23 @@ the SciPy repository::
+    Many of the steps described below can now be accomplished automatically
+    with commands which execute tasks in SciPy's Pixi workspace,
+    like ``pixi run build``.
+    To use this workspace, `install Pixi <https://pixi.sh/latest/installation/>`__
```
Probably too idiosyncratic to mention in the docs, but two issues I've had with `pixi` in an HPC context:

1) It mutates my shell profile (i.e., `~/.tcshrc`) by default, and that can mess around with `module` settings and stuff, which cascades to all kinds of other issues like causing HPC CI (https://gitlab.com/ecp-ci/jacamar-ci) to fail because it uses my mutated shell profile.
2) The install instructions linked don't work because the network is blocked; that's a bit easier to work around--I can vendor the install bash script locally and then it works.

rgommers replied:
And agreed to not mention that here.

lucascolley committed:
Merge branch 'main' into pixi-root

rgommers reviewed (approved):
Thanks @lucascolley, this now LGTM. Sorry it took a while to re-review. 

CI is a bit too red to merge it right now given how many jobs are touched here and the mix of known and new failure modes I'm seeing. I'd say once CI is green or has at most the one known Pythran/numpy-nightly failure mode, you can go ahead and merge this.

> Do you want us to hold off until you have had a chance to adjust the build dir names of scipystack to your needs? Or can I go ahead and merge?

No that's fine, I can deal with that after this PR goes in.

lucascolley committed:
add openblas for win-64 build

[skip circle]

lucascolley committed:
pin scipy-openblas32

[skip circle]
